### ISSUE TITLE
Implement partial imports

### ISSUE THREAD

**Issue by jpellizzari** [weaveworks/ui-components#5]
It would be nice to be able to do `import SomeComponent from 'weaveworks-ui-components/SomeComponent'`, instead of importing the entire module. This will allow webpack to do tree-shaking and keep bundle sizes smaller.

**Comment by jpellizzari** [comment]
This will not be possible with out-of-the-box imports with webpack. Other modules use some sort of publish step that formats the directory structure of the module. In order to do partial imports, there needs to be a directory for each sub module in the top level of the package. Here's an `ls -1` of  `material-ui` from node_modules; `lodash` follows a similar pattern:
```bash
...
RaisedButton
RefreshIndicator
SelectField
Slider
Snackbar
Stepper
Subheader
SvgIcon
Table
Tabs
TextField
TimePicker
Toggle
Toolbar
index.js
...
```

A custom resolver will need to be added in order for this to work.

**Comment by jpellizzari** [comment]
Closed by #7
